File: CSE240Assignment01/src/cse240assignment01/Guest.java
/**
 *
 * CSE 240 Assignment01
 * Program to store and manage guest information
 * @author Nikita Bahl
 */
package cse240assignment01;

public class Guest {
    String firstName;
    String lastName;
    
    //constructor to initialize firstName and lastName
    public Guest()
    {
        firstName = "???";
        lastName = "???";
    }
    
    //method to split the guest name by the '/' character
    public Guest(String guestInfo)
    {
        String[] words = guestInfo.split("/");//splits the string based on "/"
         
        this.firstName = words[0];
        this.lastName = words[1];
    }
    
    //method that returns the lastName of the guest 
    public String getLastName()
    {
        return this.lastName;
    }
    
    //method that returns the firstName of the guest
    public String getFirstName()
    {
        return this.firstName;
    }
    
    // method to return the initials of the guest in the form such as D.R.
    @Override
    public String toString()
    {
        return this.firstName.charAt(0)+ "."+this.lastName.charAt(0)+ ".";
    }
}

File: Guest.cpp
/**
 *
 * CSE 240 Assignment03
 * Program to handle guest information using user input 
 *
 * @author Nikita Bahl
 */

#include<iostream>
#include "Guest.h"
using namespace std;

      //Guest class default constructor to assign default values to each seat
     Guest::Guest()
        {
            strcpy(firstName, "???");
            strcpy(lastName , "???");
        }
      //Parameterized constructor to separate firstname and lastname frm given input  
     Guest::Guest(char* guestInfo)
        {
          char separator[2] = "/";
          char* word;
          word = strtok(guestInfo, separator);
          strcpy(firstName, word);
          word = strtok(NULL, separator);
          strcpy(lastName, word);            
        }

      //method to get last name of guest  
      char* Guest::getLastName ( )
         {
              return lastName;
         }
         
      //method to get first name of guest
       char* Guest::getFirstName ( )
         {
              return firstName;
         }
    
         //method to return initials of the guest
       char* Guest::toString ( ) 
          {
              char *initials = new char[4];
              initials[0] = firstName[0];
              initials[1] = '.';
              initials[2] = lastName[0];
              initials[3] = '.';
              
              return initials;
          }
